iothub/device/src/Transport/DefaultDelegatingHandler.cs

```diff
@@ -75,13 +75,33 @@ public virtual Task WaitForTransportClosedAsync()
 
         public virtual Task<Message> ReceiveAsync(CancellationToken cancellationToken)
         {
-            ThrowIfDisposed();
+            bool isObjectDisposedExceptionDisabled = false;
```

Comment:
This is the wrong place to perform behavior adaptation.
According to the spec: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/devdoc/architecture.md#functional-requirements the correct place is within InternalClient.